unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -165,6 +165,38 @@ public UTF8String toLowerCase() {
     return UTF8String.fromString(toString().toLowerCase());
   }
 
+  public UTF8String initCap(final byte[] byteArr) {
+    String isoString = null;
+    try {
+      isoString = new String(byteArr, "US-ASCII");
```

Comment:
@davies @rxin Can one of you decide if this is the way to go. I would still say it's an **UTF8**String and it should support unicode. As I mentioned earlier java offers an API to do this:
http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8u40-b25/java/lang/Character.java#Character.toTitleCase%28char%29